cpp/include/libxtreemfs/file_handle_implementation.h

```diff
@@ -139,6 +142,42 @@ class VoucherManager : public rpc::CallbackInterface<xtreemfs::pbrpc::OSDFinaliz
   const pbrpc::UserCredentials& user_credentials_bogus_;
 };
 
+class VoucherManagerCallback : public rpc::CallbackInterface<xtreemfs::pbrpc::OSDFinalizeVouchersResponse> {
+ public:
+  VoucherManagerCallback(VoucherManager* voucherManager,
+                         const int tryNo,
+                         const int osdCount);
+  ~VoucherManagerCallback();
```

Comment:
Mark destructor as `virtual` to make sure it gets called even when `delete`d as a `rpc::CallbackInterface` pointer.